File: frog-sample/src/main/java/com/yuluyao/frog/func/divider/DividerHorizontalFragment.java
//package com.yuluyao.frog.func.divider;
//
//import android.os.Bundle;
//import android.support.annotation.Nullable;
//import android.support.v4.app.Fragment;
//import android.support.v7.widget.LinearLayoutManager;
//import android.support.v7.widget.RecyclerView;
//import android.view.LayoutInflater;
//import android.view.View;
//import android.view.ViewGroup;
//import yuluyao.frog.decor.Divider;
//import com.yuluyao.frog.R;
//import com.yuluyao.frog.repo.Data;
//import com.yuluyao.frog.repo.Repo;
//import io.reactivex.annotations.NonNull;
//import io.reactivex.functions.Consumer;
//import java.util.List;
//
///**
// * 描 述：
// * 作 者：Vegeta Yu
// * 时 间：2017/9/5 16:33
// */
//public class DividerHorizontalFragment extends Fragment {
//
//  public static DividerHorizontalFragment newInstance() {
//    Bundle args = new Bundle();
//    DividerHorizontalFragment fragment = new DividerHorizontalFragment();
//    fragment.setArguments(args);
//    return fragment;
//  }
//
//
//  @Nullable @Override
//  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
//      @Nullable Bundle savedInstanceState) {
//    return inflater.inflate(R.layout.fragment_divider_horizontal, container, false);
//  }
//
//  @Override public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
//    super.onViewCreated(view, savedInstanceState);
//    RecyclerView recyclerView = (RecyclerView) view.findViewById(R.id.recycler);
//    recyclerView.setLayoutManager(
//        new LinearLayoutManager(getContext(), LinearLayoutManager.HORIZONTAL, false));
//    recyclerView.addItemDecoration(new Divider(2, R.color.colorAccent));
//
//    final DividerHorizontalAdapter adapter = new DividerHorizontalAdapter();
//    recyclerView.setAdapter(adapter);
//
//    Repo.getInstance(getContext()).refresh().subscribe(new Consumer<List<Data>>() {
//      @Override public void accept(@NonNull List<Data> datas) throws Exception {
//        adapter.notifyRefreshCompleted(datas);
//      }
//    });
//
//  }
//
//}

File: frog-sample/src/main/java/com/yuluyao/frog/func/divider/DividerHorizontalFragment.kt
package com.yuluyao.frog.func.divider

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import yuluyao.frog.decor.Divider
import com.yuluyao.frog.R
import com.yuluyao.frog.repo.Data
import com.yuluyao.frog.repo.Repo
import io.reactivex.annotations.NonNull
import io.reactivex.functions.Consumer
import kotlinx.android.synthetic.main.fragment_divider_horizontal.*
import yuluyao.frog.FrogAdapter

/**
 * 描 述：
 * 作 者：Vegeta Yu
 * 时 间：2017/9/5 16:33
 */
class DividerHorizontalFragment : Fragment() {
  val adapter = FrogAdapter<Data>(R.layout.item_data_horizontal)

  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                            savedInstanceState: Bundle?): View? {
    return inflater.inflate(R.layout.fragment_divider_horizontal, container, false)
  }

  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)
    recycler.layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
    recycler.addItemDecoration(Divider(2f, R.color.item_decoration))

    recycler.adapter = adapter

    Repo.refresh().subscribe {
      adapter.data = it
      adapter.notifyDataSetChanged()
    }

  }

//  companion object {
//
//    fun newInstance(): DividerHorizontalFragment {
//      val args = Bundle()
//      val fragment = DividerHorizontalFragment()
//      fragment.arguments = args
//      return fragment
//    }
//  }

}
